assets/cluster-bootstrap/cluster-ingresscontrollers-02-config.yaml

```diff
@@ -4,12 +4,15 @@ metadata:
   name: default
   namespace: openshift-ingress-operator
 spec:
-{{ if .EndpointPublishingStrategyScope }}
+{{- if eq .RouterServiceType "Private" }}
+  endpointPublishingStrategy:
+    type: Private 
+{{- else if .EndpointPublishingStrategyScope }}
```

Comment:
I think that I follow the plan here.  Basically `RouterServiceType` can be either `Private` or `LoadBalancer` and `EndpointPublishingStrategyScope` is used when `RouterServiceType` is `LoadBalancer`.  Given this, it seems best if we cleanup the interface here.  I propose changing `{{- else if .EndpointPublishingStrategyScope }}` to `{{- else if eq .RouterServiceType "LoadBalancer" }}`.  So if an LB router is requested, `EndpointPublishingStrategyScope` must be set.  What do you think?

Reply:
I agree, that would be a much cleaner logic here. The reason for the current setup is that I wanted to make the least change necessary. In order not to break anything.
I believe you have a much better understanding on how the new setup you proposed would apply for example when existing clusters are upgraded. If you are OK with the setup you proposed, I am fine, too, and I can update it accordingly.

Reply:
Updated the yaml. Let's see if it meets your expectations. I kept `EndpointPublishingStrategyScope` optional, as it is optional in the API definition, too.

Reply:
I like what you've done here.  It is cleaner and is backwards compatible with what we have today.